### PR TITLE
Issue/pkg 599

### PR DESCRIPTION
**Pull request by ryanskeith**
After reviewing the code, I added some more verbosity to the tests and also created a new script to see the full affects of the hotfixes so it could be seen in a more declarative manner and we could catch unintended lasting changes.

### CODE REVIEW

**Review by cbouss — APPROVED** [review]
LGTM!

**Review by danpetry — COMMENTED** [review]
Overall really nice and useful script. There's a ton of tech debt here and it would be great to clean it up.

Functionally nice, but I've highlighted code cleanliness a lot here. Because it seems that it would be helpful to edit the script in future to get more analytics out of the hotfixes, to see where to focus attention in general, or perhaps to run different kinds of queries to see what hotfixes could be removed for certain packages while we're releasing new versions.

Also could you please add a test for the script to the ci.yaml and address the flake8 lints?

File: gen-current-hotfix-report.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+from collections import defaultdict
+from pprint import pprint
+from pathlib import Path
+
+
+from conda.exports import subdir as conda_subdir
+from conda_build.index import _apply_instructions
+import urllib
+
+html_differ = difflib.HtmlDiff()
```
[the remark this answers is not in the document]
Comment (ryanskeith) [reply gen-current-hotfix-report.py:14]: True, but on second glance, I don't need this code.

File: gen-current-hotfix-report.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    channel_subdir_dir = Path(channel) / subdir
+
+    # out_file =  channel_subdir_dir / "repodata-reference.json"
+    # url = f"{channel_base_url}/{subdir}/repodata.json"
+    # print(f"Downloading 'repodata.json' from {url}")
+    # urllib.request.urlretrieve(url, out_file)
+
+    out_file = channel_subdir_dir /  "repodata_from_packages.json"
+    url = f"{channel_base_url}/{subdir}/repodata_from_packages.json"
+    print(f"Downloading 'repodata_from_packages.json' from {url}")
```
[the remark this answers is not in the document]
Comment (ryanskeith) [reply gen-current-hotfix-report.py:59]: This is true.  I wonder though if it is worth the cost for such simple scripts.  I think we need to define what we want repodata-hotfixes to do for us in the future.

**Pull request merged**